src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentials.java

```diff
@@ -45,6 +45,7 @@
     String getDisplayName();
 
     AWSCredentials getCredentials(String mfaToken);
+    AWSCredentials getCredentials(int stsTokenDuration);
```

Comment:
Adding a new method is a binary incompatible change. Can you check if there are any implementations (at least in the `jenkinsci` organization) that could break? Anyway, some kind of notice would be needed.

Reply:
Hey @andresrc, I've just been thinking about this lately, what do you think if this change was removed, and the dependent class (`AmazonWebServicesCredentialsBinding`) was adjusted to reference our class rather than this interface?

As its usage is within the same package, I think this should be okay; could you please validate this? I'm not 100% sure on what's coming in through the `build` parameter, however, and not sure if this would influence whether the resulting var will always be of `AWSCredentialsImpl` type.

I haven't actually tested this theory, so it might not even functionally work, but just your thoughts on the approach before trying it.

---

Location: [found here](https://github.com/jenkinsci/aws-credentials-plugin/blob/7e201fb29119de649829c38edb33eb2bfe9b5c9c/src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java#L108)

```java
        AmazonWebServicesCredentials credentialRetriever = getCredentials(build);
```

would become

```java
        AWSCredentialsImpl credentialRetriever = getCredentials(build);
```